### PR TITLE
feat(client): link production page with machine and scheduling pages

### PR DESCRIPTION
**Pull request by SiuYe**
#

## Changes this PR introduces

- **assembly machine** input from production modal is now a select from machine available
- scheduling modal **partType** now takes any string which is the **name** of the production create

## How to test this PR

- create a part and a bike through the production page and assure that it's updated on the scheduling page

## Notes for reviewers

- don't be shy to tell me to redo anything, I will gladly change it

## Anything you'd like the reviewers to be aware of

- price of a machinery is now 1-to-1 with the quantity created
- machine work every day

## Checklist

- [x] I renamed the title of my pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-rh804j Link production page with machine and scheduling pages](https://app.clickup.com/t/rh804j)

**Review by gordonpn — COMMENTED** [review]
PR looks good to me!

But can you make some modifications to the production page?

1. Change this button text to say `Change production type` rather than `Bike`/`Part`

image

2. Make modals close after submitting successfully.

File: client/src/hooks/useProductionModal.jsx [hunk 1]
```diff
@@ -171,6 +174,18 @@ const useProductionModal = () => {
           note: noteRef.current.value,
         });
       }
+      // update scheduling page
+      await createScheduling({
+        partType: nameRef.current.value,
```
Comment (Mackbellemore) [inline client/src/hooks/useProductionModal.jsx:179]: somethings wrong with this, 
image

File: server/app/entities/Scheduling.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
 import { BadRequestError } from '../errors';
 import Joi from 'joi';
-import { PartTypes } from './Part';
 import { IScheduling } from '../models/SchedulingModel';
 
 export default class SchedulingEntity {
   // Optional schema to validate Patch requests
   private static optionalSchema = Joi.object().keys({
-    partType: Joi.string().valid(...PartTypes),
```
Comment (Mackbellemore) [inline server/app/entities/Scheduling.ts:9]: probably related ^

Comment (Mackbellemore) [reply server/app/entities/Scheduling.ts:9]: actually this is definitely it

File: client/src/hooks/useProductionModal.jsx (hunk 1 above)
Comment (Mackbellemore) [reply client/src/hooks/useProductionModal.jsx:179]: nevermind

**Review by Mackbellemore — DISMISSED** [review]
LGTM!, make your backend changes seperate PR next time so i dont have a brainfart, Ill re approve after gordos requests

**Comment by SiuYe** [comment]
> PR looks good to me!
> 
> But can you make some modifications to the production page?
> 
> 1. Change this button text to say `Change production type` rather than `Bike`/`Part`
> 
> image
> 
> 1. Make modals close after submitting successfully.

It has been updated!!

**Comment by SiuYe** [comment]
> LGTM!, make your backend changes seperate PR next time so i dont have a brainfart, Ill re approve after gordos requests

totally agree with what you are saying @Mackbellemore , doing a separate PR for the server would have been better! but there was only 1 line of code change, so I slid it in quick quick 👀 and it might have looked sus if there was only 1 line of code changed on a PR.

reapprove time? if everything is good with @gordonpn

**Review by gordonpn — APPROVED** [review]
LGTM! 🚀🚀🚀

**Review by Mackbellemore — APPROVED** [review]

**Pull request merged**